pennylane/measurements/classical_shadow.py

```diff
@@ -291,19 +389,20 @@ def wires(self):
     def queue(self, context=qml.QueuingManager):
         """Append the measurement process to an annotated queue, making sure
         the observable is not queued"""
-        if self.H is not None:
-            Hs = [self.H] if not isinstance(self.H, Iterable) else self.H
-            for H in Hs:
-                context.update_info(H, owner=self)
-            context.append(self, owns=Hs)
-        else:
-            context.append(self)
+        Hs = self.H if isinstance(self.H, Iterable) else [self.H]
+        for H in Hs:
+            context.update_info(H, owner=self)
+        context.append(self, owns=Hs)
 
         return self
 
     def __copy__(self):
-        obj = super().__copy__()
-        obj.seed = self.seed
-        obj.H = self.H
-        obj.k = self.k
-        return obj
+        return self.__class__(
+            self.return_type,
+            H=self.H,
```

Comment:
I think we should copy all the observables in `H` too. At the moment there are no issues as it is, but it would be good to copy just in case.

Reply:
Good catch!